parser_library/src/workspaces/workspace.cpp

```diff
@@ -388,62 +382,78 @@ const processor_group& workspace::get_proc_grp(const proc_grp_id& proc_grp) cons
     return proc_grps_.at(proc_grp);
 }
 
-void replace_all(std::string& str, std::string_view what, std::string_view with)
+namespace {
+std::pair<size_t, char> match_windows_uri_with_drive(const std::string& input)
 {
-    for (std::string::size_type p = 0; (p = str.find(what, p)) != std::string::npos; p += with.size())
-        str.replace(p, what.size(), with);
+    if (std::smatch matches; std::regex_search(input, matches, std::regex("^file:///([a-zA-Z])(?::|%3[aA])")))
+        return { matches[0].length(), matches[1].str()[0] };
+    else
+        return { 0, '\0' };
 }
 
-bool starts_with(std::string_view s, std::string_view b) { return s.substr(0, b.size()) == b; }
+size_t preprocess_uri_with_windows_drive_letter_regex_string(const std::string& input, std::string& r)
+{
+    auto [match_length, drive_letter] = match_windows_uri_with_drive(input);
+
+    if (match_length == 0)
+        return 0;
 
-std::regex pathmask_to_regex(std::string input)
+    // Append windows file path (e.g. ^file:///[cC](?::|%3[aA]))
+    r.append("^file:///[");
+    r.push_back(static_cast<char>(tolower(drive_letter)));
+    r.push_back(static_cast<char>(toupper(drive_letter)));
+    r.append("](?::|%3[aA])");
+
+    return match_length;
+}
+} // namespace
+std::regex pathmask_to_regex(const std::string& input)
 {
     std::string r;
     r.reserve(input.size());
 
-    replace_all(input, "\\", "/");
-    if (input.back() != '/')
-        input.push_back('/');
-
     std::string_view s = input;
 
+    // URI mask shouldn't care about Windows Drive letter case
+    s.remove_prefix(preprocess_uri_with_windows_drive_letter_regex_string(input, r));
+
     bool path_started = false;
     while (!s.empty())
     {
         switch (auto c = s.front())
         {
             case '*':
-                if (starts_with(s, "**/"))
+                if (s.starts_with("**/"))
                 {
                     if (path_started)
                     {
                         path_started = false;
-                        r.append("[^/\\\\]*[/\\\\]");
+                        r.append("[^/]*[/]");
                     }
-                    r.append("(?:[^/\\\\]+[/\\\\])*");
+                    r.append("(?:[^/]+[/])*");
```

Comment:
If we allow the `//` pattern in the path, we could replace this part with simpler pattern and perhaps avoid the msvc implementation issue.
```suggestion
                    r.append("(?:.*/)?");
```